server/davidia/server/processor.py

```diff
@@ -1,17 +1,39 @@
 from __future__ import annotations
 
-import random
-
-from ..models.messages import (AppendLineDataMessage, AxesParameters,
-                               ClearSelectionsMessage, ClientSelectionMessage,
-                               HeatmapData, ImageData, ImageDataMessage, LineData,
-                               MsgType, MultiLineDataMessage, PlotMessage, ScatterData,
-                               ScatterDataMessage, SelectionsMessage, SurfaceData,
-                               SurfaceDataMessage, TableData, TableDataMessage,
-                               UpdateSelectionsMessage)
+from ..models.messages import (
+    AppendLineDataMessage,
+    AxesParameters,
+    ClearSelectionsMessage,
+    ClientSelectionMessage,
+    ClientLineParametersMessage,
+    HeatmapData,
+    ImageData,
+    ImageDataMessage,
+    LineData,
+    LineParams,
+    MsgType,
+    MultiLineDataMessage,
+    PlotMessage,
+    ScatterData,
+    ScatterDataMessage,
+    SelectionsMessage,
+    SurfaceData,
+    SurfaceDataMessage,
+    TableData,
+    TableDataMessage,
+    UpdateSelectionsMessage,
+)
 from ..models.selections import as_selection
 
 
+def check_line_names(lines: list[LineData]) -> list[LineData]:
+    """Autonames lines that do not have names"""
+    for index, line in enumerate(lines):
+        if not line.line_params.name:
+            line.line_params.name = f"Line {index}"
```

Comment:
Updated so that it will increment from index if that anmes already exists. Although, alternatively could start at "Line 0" for each and use the first number that hasn't already been used.